File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int consecutiveSum = 0;
        int ans = INT_MIN;
        for(int i= 0 ; i < nums.size() ; i++){
            consecutiveSum+=nums[i];
            if(consecutiveSum > ans){
                ans = consecutiveSum;
            }
            if(consecutiveSum < 0){
                consecutiveSum = 0;
            }
        }
        return ans;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cSum = 0;
        int ans = Integer.MIN_VALUE;
        for(int i = 0 ; i < nums.length ; i++){
            cSum+=nums[i];
            if(cSum > ans){
                ans = cSum;
            }
            if(cSum < 0){
                cSum = 0;
            }
        }
        return ans;
    }
}